lib/foreman_inventory_upload/async/generate_report_job.rb

```diff
@@ -18,8 +19,9 @@ def plan(base_folder, organization_id)
             base_folder,
             ForemanInventoryUpload.facts_archive_name(organization_id),
             organization_id
-          )
+          ) unless disconnected
```

Comment:
This means that the action will not be executed at all if `disconnected` is set to `false`